src/Azure.Functions.Cli/Helpers/PowerShellHelper.cs

```diff
@@ -0,0 +1,100 @@
+﻿using System;
+using System.IO;
+using System.Threading.Tasks;
+using Azure.Functions.Cli.Common;
+using System.Net.Http;
+using System.Xml;
+
+namespace Azure.Functions.Cli.Helpers
+{
+    public static class PowerShellHelper
+    {
+        // The module name for the Azure Resource Manager.
+        private const string AzModuleName = "Az";
+
+        // The PowerShellGallery uri to query for the latest module version.
+        private const string PowerShellGalleryFindPackagesByIdUri = "https://www.powershellgallery.com/api/v2/FindPackagesById()?id=";
+        
+        /// <summary>
+        /// Gets the latest supported major version of the Az module from the PSGallery.
+        /// </summary>
+        public static async Task<string> GetLatestAzModuleMajorVersion()
+        {
+            Uri address = new Uri($"{PowerShellGalleryFindPackagesByIdUri}'{AzModuleName}'");
+            int maxNumberOfTries = 3;
+            int tries = 1;
+
+            string latestMajorVersion = null;
+            Stream stream = null;
+            using (HttpClient client = new HttpClient())
+            {
+                while (tries <= maxNumberOfTries)
```

Comment:
is it possible to use `Helpers.RetryHelper` here?

Reply:
Thanks for pointing this out. Yes, we can. I have fixed the code accordingly.